pkg/ccl/streamingccl/streamclient/span_config_stream_client.go

```diff
@@ -13,87 +13,98 @@ import (
 	"net/url"
 
 	"github.com/cockroachdb/cockroach/pkg/ccl/streamingccl"
-	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
-	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/errors"
 	"github.com/jackc/pgx/v4"
 )
 
-type spanConfigStreamClient struct {
+// SpanConfigClient provides methods to interact with a stream of span
+// config updates of a specific application tenant.
+type SpanConfigClient interface {
+	Client
+
+	// SetupSpanConfigsStream creates a stream for the span configs
+	// that apply to the passed in tenant, and returns the subscriptions the
+	// client can subscribe to. No protected timestamp or job is persisted to the
+	// source cluster.
+	SetupSpanConfigsStream(tenant roachpb.TenantName) (Subscription, error)
+}
+type spanConfigClient struct {
 	pgxConfig    *pgx.ConnConfig
 	srcConn      *pgx.Conn // pgx connection to the source cluster
 	subscription spanConfigStreamSubscription
 }
 
-// TODO(msbutler): once the span config stream is hooked up on the ingestion
-// side, consider implementing a different interface for the spanConfig client.
-var _ Client = &spanConfigStreamClient{}
+var _ SpanConfigClient = &spanConfigClient{}
 
 func NewSpanConfigStreamClient(
-	ctx context.Context, remote *url.URL, opts ...Option,
-) (Client, error) {
-	options := processOptions(opts)
+	ctx context.Context, remote *url.URL, db isql.DB,
+) (SpanConfigClient, error) {
+
+	if remote.Scheme == "external" {
+		if db == nil {
+			return nil, errors.AssertionFailedf("nil db handle can't be used to dereference external URI")
+		}
+		addr, err := lookupExternalConnection(ctx, remote.Host, db)
+		if err != nil {
+			return nil, err
+		}
+		url, err := addr.URL()
+		if err != nil {
+			return nil, err
+		}
+		return NewSpanConfigStreamClient(ctx, url, db)
+	}
 
-	config, err := setupPGXConfig(remote, options)
+	config, err := setupPGXConfig(remote, &options{})
```

Comment:
Can we still process options here? One option is the StreamID which would eventually be nice to thread through to this call as well so that we get the appropriate application name on the connection.

Reply:
Done.